src/main/python/camdkit/model.py

```diff
@@ -44,63 +60,768 @@ class ISO(StrictlyPositiveIntegerParameter):
+      if test != None and not (isinstance(test, float) and test >= 0.0 and test <= 1.0):
+        return False
+    return True
+
+  @staticmethod
+  def to_json(value: typing.Any) -> typing.Any:
+    return {k: v for k, v in dataclasses.asdict(value).items() if v is not None}
+
+  @staticmethod
+  def to_pretty_json(value: typing.Any) -> typing.Any:
+    d = LensEncoders.to_json(value)
+    for k,v in d.items():
+      d[k] = round(v, PRETTY_FLOAT_DP)
+    return d
+  
+  @staticmethod
+  def from_json(value: typing.Any) -> typing.Any:
+    return Encoders(**value)
+
+  @staticmethod
+  def make_json_schema() -> dict:
+    return {
+      "type": "object",
+      "additionalProperties": False,
+      "properties": {
+        "focus": {
+          "type": "number",
+          "minimum": 0.0,
+          "maximum": 1.0
+        },
+        "iris": {
+          "type": "number",
+          "minimum": 0.0,
+          "maximum": 1.0
+        },
+        "zoom": {
+          "type": "number",
+          "minimum": 0.0,
+          "maximum": 1.0
+        }
+      },
+      "anyOf": [ {"required": ["focus"]}, {"required": ["iris"]}, {"required": ["zoom"]}
+      ]
+    }
+  
+class LensRawEncoders(Parameter):
+  """
+  Raw encoder values for focus, iris and zoom.
+  These values are dependent on encoder resolution and before any homing / ranging has taken place.
+  """
+  sampling = Sampling.REGULAR
+  canonical_name = "rawEncoders"
+  section = "lens"
+  units = None
+
+  @staticmethod
+  def validate(value) -> bool:
+    """
+    The parameter shall contain at least one integer value for the FIZ encoders.
+    """
+    if not isinstance(value, RawEncoders):
+      return False
+    if value.focus == None and value.iris == None and value.zoom == None:
+      return False
+    for test in [value.focus, value.iris, value.zoom]:
+      if test != None and not (isinstance(test, int) and test >= 0):
+        return False
+    return True
+
+  @staticmethod
+  def to_json(value: typing.Any) -> typing.Any:
+    return {k: v for k, v in dataclasses.asdict(value).items() if v is not None}
+
+  @staticmethod
+  def to_pretty_json(value: typing.Any) -> typing.Any:
+    d = LensRawEncoders.to_json(value)
+    for k,v in d.items():
+      d[k] = round(v, PRETTY_FLOAT_DP)
+    return d
+  
+  @staticmethod
+  def from_json(value: typing.Any) -> typing.Any:
+    return RawEncoders(**value)
+
+  @staticmethod
+  def make_json_schema() -> dict:
+    return {
+      "type": "object",
+      "additionalProperties": False,
+      "properties": {
+        "focus": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "iris": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "zoom": {
+          "type": "integer",
+          "minimum": 0
+        }
+      },
+      "anyOf": [ {"required": ["focus"]}, {"required": ["iris"]}, {"required": ["zoom"]}
+      ]
+    }
+  
+  
+class TimingMode(EnumParameter):
```

Comment:
Isn't this a semantically equivalent copy of what's at 473:482 (with the only difference being the wording of the doctoring)?

Reply:
Please can you clarify - I think the line numbers have changed since this comment. Thanks!